ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -241,6 +248,12 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			break
 		}
 
+		if rayJobInstance.Spec.SubmissionMode == rayv1.SidecarMode {
+			if shouldUpdate := r.checkSidecarContainerAndUpdateStatusIfNeeded(ctx, rayJobInstance); shouldUpdate {
+				break
+			}
+		}
+
 		job := &batchv1.Job{}
```

Comment:
* `job := &batchv1.Job{}` => `isJobFinished := true`

```go
func checkSubmitterAndUpdateStatusIfNeeded(submissionMode, ...) {
   // handle K8sJobMode
   // handle SidecarMode
   return isJobFinished
}

```

Then, we can remove:

```go
		// If in K8sJobMode, further refine the terminal condition by checking if the submitter Job has finished.
		// See https://github.com/ray-project/kuberay/pull/1919 for reasons.
		if rayJobInstance.Spec.SubmissionMode == rayv1.K8sJobMode {
			_, finished := utils.IsJobFinished(job)
			isJobTerminal = isJobTerminal && finished
		}
```